File: solutions/ocaml/05-iz3/code/src/main.ml
let starts_with s prefix =
  let len_s = String.length s in
  let len_p = String.length prefix in
  len_s >= len_p && String.sub s 0 len_p = prefix

let () =
  while true do
    print_string "$ ";
    flush stdout;
    let command = input_line stdin in
    if command = "exit" then
      exit 0
    else if starts_with command "echo " then
      print_endline (String.sub command 5 (String.length command - 5))
    else
      print_endline (command ^ ": command not found")
  done

File: solutions/zig/05-iz3/code/src/main.zig
const std = @import("std");

pub fn main(init: std.process.Init) !void {
    var stdin_buffer: [4096]u8 = undefined;
    var stdin = std.Io.File.stdin().readerStreaming(init.io, &stdin_buffer);
    var stdout = std.Io.File.stdout().writer(init.io, &.{});

    while (true) {
        try stdout.interface.print("$ ", .{});
        const command = try stdin.interface.takeDelimiter('\n');

        if (std.mem.eql(u8, command.?, "exit")) {
            break;
        } else if (std.mem.startsWith(u8, command.?, "echo ")) {
            try stdout.interface.print("{s}\n", .{command.?[5..]});
        } else {
            try stdout.interface.print("{s}: command not found\n", .{command.?});
        }
    }
}
